src/callbacks/callbacks.jl

```diff
@@ -346,6 +415,22 @@ function set_insolation_variables!(Y, p, t, ::GCMDrivenInsolation)
         Fields.field2array(p.external_forcing.insolation)
 end
 
+function set_insolation_variables!(Y, p, t, ::ExternalTVInsolation)
+    # unpack objects with time varying data
+    (; rrtmgp_model) = p.radiation
+    (; coszen, rsdt) = p.external_forcing.surface_inputs
+    coszen_tv =
+        getproperty(p.external_forcing.surface_timevaryinginputs, :coszen)
```

Comment:
```suggestion
    coszen_tv =
       p.external_forcing.surface_timevaryinginputs.coszen
```
Why is this not just written like this? (same below)

Reply:
Changed this - you're right its simpler and consistent with what I do elsewhere - thanks @charleskawczynski !